tests/model_preparation/test_summaries.py

```diff
@@ -0,0 +1,240 @@
+import os
+import json
+import pandas as pd
+import random
+import string
+from unittest import TestCase
+from tempfile import TemporaryDirectory
+from hypothesis import (
+    given,
+    settings,
+)
+from hypothesis.strategies import (
+    just,
+    integers,
+)
+
+from oasislmf.manager import OasisManager as om
+from oasislmf.model_preparation.summaries import write_exposure_summary
+from oasislmf.model_preparation.gul_inputs import get_gul_input_items
+from oasislmf.utils.coverages import SUPPORTED_COVERAGE_TYPES
+from oasislmf.utils.status import OASIS_KEYS_STATUS
+from oasislmf.utils.profiles import get_oed_hierarchy
+from oasislmf.utils.peril import PERILS
+
+from tests.data import (
+    source_exposure,
+    write_source_files,
+    write_keys_files,
+)
+
+
+class TestSummaries(TestCase):
+
+    @staticmethod
+    def get_keys_dict(peril_ls, nlocations, successful=True, start_loc=1):
```

Comment:
@hchagani-oasislmf This isn't my full review (which I will do on Monday), but I note that this method (`get_keys_dict`) is unnecessary. So we already have exposure and keys data generators which use the Hypothesis library in the `tests/data.py` module

https://github.com/OasisLMF/OasisLMF/blob/develop/tests/data.py

This is the keys data generator (generates a list of keys dicts)

https://github.com/OasisLMF/OasisLMF/blob/develop/tests/data.py#L354

and you can see it being used in the FM acceptance tests module

https://github.com/OasisLMF/OasisLMF/blob/develop/tests/acceptance/test_fm_acceptance.py#L108

This is a locations data generator

https://github.com/OasisLMF/OasisLMF/blob/develop/tests/data.py#L197

and you can see it also being used in the FM tests module. 

There are also methods that will write a list of keys records to a keys file).

Reply:
I wanted to create keys data where multiple perils were associated with a single location. I couldn't see a way to do this using the keys data generator in `tests/data.py`.

Potentially, I could add a method in `tests/data.py` to do this instead of the `get_keys_dict` method in `tests/model_preparation/test_summaries.py`.

Reply:
You can probably re-use the `keys` method in `data.py` - just generate a list of items with temporary values for peril ID and then in the test case modify the dicts as you need at the top of the test case, before processing. We do something similar in the FM acceptance tests.

You can do the same thing for the locations.

If you can't make this work then of course you can use your custom method. But I think the general preference should to re-use existing data generators, which were designed to be customised.